servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/AbstractHttpServiceAsyncContextTest.java

```diff
@@ -326,25 +337,139 @@ final void connectionAcceptorContextDoesNotLeak(HttpProtocol protocol, boolean u
                                                     ConnectionAcceptorType connectionAcceptorType) throws Exception {
         Assumptions.assumeFalse(isBlocking() && useImmediate, "Blocking service can only run with offloading");
 
+        Queue<Throwable> errorQueue = new ConcurrentLinkedQueue<>();
+        AtomicReference<ContextMap> currentContextMap = new AtomicReference<>();
         HttpServerBuilder builder = HttpServers.forAddress(localAddress(0)).protocols(protocol.config);
         switch (connectionAcceptorType) {
             case EARLY:
-                builder.appendEarlyConnectionAcceptor(conn -> {
-                    AsyncContext.put(K1, "v1");
+                builder.appendEarlyConnectionAcceptor(newEarly(useImmediate, conn -> {
+                    AsyncContext.put(K1, "early");
```

Comment:
are these puts (for early, late and deprecated) asserted somewhere? I see you did it for ALL and ALL_TWICE in the succeeding ones, but here there is only one. I see `assertAsyncContext` calls using the key K1, but they only assert on a request ID but I am not sure how it aligns with these cases.

Reply:
The idea of original tests was to check that whatever you put in acceptor does not leak to request controller. There is an assertion in `newEmptyAsyncContextService()` implementations, see `serverWithEmptyAsyncContextService(...)` in subclasses